src/utils.jl

```diff
@@ -107,3 +130,81 @@ end
   isnothing(ret) && return IteratorReturn(nothing)
   ret
 end
+
+# this is similar to code_typed but it considers the world age
+function code_typed_by_type(@nospecialize(tt::Type);
+                            optimize::Bool=true,
+                            world::UInt=Base.get_world_counter(),
+                            interp::Core.Compiler.AbstractInterpreter=Core.Compiler.NativeInterpreter(world))
+    tt = Base.to_tuple_type(tt)
+    # look up the method
+    match, valid_worlds = Core.Compiler.findsup(tt, Core.Compiler.InternalMethodTable(world))
+    meth = Base.func_for_method_checked(match.method, tt, match.sparams)
+    # run inference, normally not allowed in generated functions
+    frame = Core.Compiler.typeinf_frame(interp, meth, match.spec_types, match.sparams, optimize)
```

Comment:
```suggestion
    # run inference, normally not allowed in generated functions
    frame = Core.Compiler.typeinf_frame(interp, match.method, match.spec_types, match.sparams, optimize)
```
Now Julia base removed `func_for_method_checked` (xref: https://github.com/JuliaLang/julia/pull/51444)

Reply:
ah that was about the last remnant of blindly copied `code_typed_by_type` I think